src/foam/nanos/crunch/CapabilityAvailabilityDAO.js

```diff
@@ -51,8 +55,13 @@ foam.CLASS({
     {
       name: 'find_',
       javaCode: `
+        Theme theme = (Theme) x.get("theme");
+        List<String> restrictedCapabilities = new ArrayList<String>();
+        if ( theme != null && theme.getRestrictedCapabilities() != null ) {
+          restrictedCapabilities = new ArrayList(Arrays.asList(theme.getRestrictedCapabilities()));
```

Comment:
This approach isn't very efficient because you're doing an 2XO(n) contains() and you're interrogating the Theme's internal state, rather than just asking it to tell you what you really want to know. It would be better if Theme and a method called boolean restrictsCapability(capability), and then it could implement it however it wanted, including by having a transient hashSet which is created from restrictedCapabilities but lets you test in O(1) instead.